projects/js-packages/components/components/threat-modal/index.tsx

```diff
@@ -65,88 +65,38 @@ export default function ThreatModal( {
 		return getFixerState( threat.fixer );
 	}, [ threat.fixer ] );
 
-	const getModalTitle = useMemo( () => {
-		if ( userConnectionNeeded ) {
-			return <Text variant="title-small">{ __( 'User connection needed', 'jetpack' ) }</Text>;
-		}
-
-		if ( siteCredentialsNeeded ) {
-			return <Text variant="title-small">{ __( 'Site credentials needed', 'jetpack' ) }</Text>;
-		}
-
-		return (
-			<>
-				<Text variant="title-small">{ threat.title }</Text>
-				{ !! threat.severity && <ThreatSeverityBadge severity={ threat.severity } /> }
-			</>
-		);
-	}, [ userConnectionNeeded, siteCredentialsNeeded, threat.title, threat.severity ] );
-
 	return (
 		<Modal
+			title={
+				<div className={ styles.title }>
+					<Text variant="title-small">{ threat.title }</Text>
+					{ !! threat.severity && <ThreatSeverityBadge severity={ threat.severity } /> }
+				</div>
+			}
 			size="large"
-			title={ <div className={ styles.title }>{ getModalTitle }</div> }
 			{ ...modalProps }
 		>
 			<div className={ styles[ 'threat-details' ] }>
-				<UserConnectionGate
-					userConnectionNeeded={ userConnectionNeeded }
-					userIsConnecting={ userIsConnecting }
-					handleConnectUser={ handleConnectUser }
+				<ThreatModalContext.Provider
+					value={ {
+						closeModal: modalProps.onRequestClose,
+					} }
 				>
-					<CredentialsGate
+					<ThreatFixConfirmation
+						threat={ threat }
+						fixerState={ fixerState }
```

Comment:
We're passing both the `threat` and `fixerState` into `ThreatFixConfirmation` - the `fixerState` is only used inside the component, and computed from the `threat`, so we could remove the computation of `fixerState` in this file and move it into `ThreatFixConfirmation` as necessary.